metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java

```diff
@@ -506,20 +536,44 @@ public void ingestEntity(@Nonnull Entity entity, @Nonnull AuditStamp auditStamp,
             .collect(Collectors.toList())));
   }
 
+  /**
+  Returns true if entityType should have some aspect as per its definition
+    but aspects given does not have that aspect
+   */
+  private boolean shouldAddAspect(String entityType, String aspectName, Set<String> aspects) {
```

Comment:
The name of this method is a bit strange. The only case in which we should inject an aspect artificially is for a special aspect like BrowsePaths. This method name makes it seem like it applies to any aspect that the entity is associated with, but is not provided

Reply:
Any other suggestions for names? This looked like a long piece of code so I extracted it out. Maybe `isValidAspectMissing` better?